### PR TITLE
Added Boolean algebra Cheat Sheet 

### PR DESCRIPTION
**Pull request by gautamkrishnar**
**What does your Instant Answer do?**
It will display all basic equations of Boolean algebra as cheat sheet

**What is the data source for your Instant Answer? (Provide a link if possible)**
(n/a)

**Why did you choose this data source?**
(n/a)

**Are there any other alternative (better) data sources?**
(n/a)

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Students

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/6230/logic-cheat-sheet

**Are you having any problems? Do you need our help with anything?**
Just started contributing... Its my first pull request :)

**Where did you hear about DuckDuckHack? (For first time contributors)**
When i was wandering online, I found it, I fell in love with it.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screenshot 423

### CODE REVIEW

File: share/goodie/cheat_sheets/json/logic.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
+{
+  "id": "logic_cheat_sheet",
+  "name": "logic",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/logic.json]: Let's change this to "Boolean Algebra" or "Boolean Algrebra Logic" -- something a little more descriptive/specific

File: share/goodie/cheat_sheets/json/logic.json [hunk 2]
```diff
@@ -0,0 +1,172 @@
+{
+  "id": "logic_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/logic.json]: similarly, I think this should be `boolean_algebra_cheat_sheet`

File: share/goodie/cheat_sheets/json/logic.json [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  "name": "logic",
+  "description": "These are a set of operators and expressions used in the boolean algebra",
+  "metadata": {
+    "sourceName": "Image",
+    "sourceUrl": "http://i.imgur.com/OXdpVxx.png"
+  },
+  "aliases": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/logic.json]: :+1:

File: share/goodie/cheat_sheets/json/logic.json [hunk 4]
```diff
@@ -0,0 +1,172 @@
+{
+  "id": "logic_cheat_sheet",
+  "name": "logic",
+  "description": "These are a set of operators and expressions used in the boolean algebra",
+  "metadata": {
+    "sourceName": "Image",
+    "sourceUrl": "http://i.imgur.com/OXdpVxx.png"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/logic.json]: Any chance we can find a slightly better/more-usable source? Does https://en.wikipedia.org/wiki/Boolean_algebra cover everything?

File: share/goodie/cheat_sheets/json/logic.json (hunk 3 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/logic.json]: We also now have support for a few CheatSheet templates. Let's try adding `"template_type": "reference"` which should make this easier to read. Alternatively the `code` template may also work well.

**Comment by moollaza** [comment]
@gautamkrishnar thanks a lot for contributing! We really appreciate it :smile: 

@abeyang this one is a new edge case -- a lot of the "keys" don't really have a value or explanation. I've asked the dev to switch this to the `reference` template and I think the final result should still look OK.

Any thoughts on the content? I think we should probably space out the math a little so the letters and symbols aren't all squished together?

**Comment by gautamkrishnar** [comment]
Added all changes...
Thanks for your support...

**Comment by moollaza** [comment]
@gautamkrishnar whoops, it looks like you have conflicts on your PR.

Please update it so that the only file being added is the single .json file for your Cheat Sheet. We now have an `"aliases" : ...` property which you can use to provide an array of additional trigger words.

In this case for "boolean-algebra.json" you would want to add `"aliases": "boolean logic", "logic" ...`

**Comment by gautamkrishnar** [comment]
Moved to new pull request:
https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1424
@moollaza @abeyang

File: share/goodie/cheat_sheets/json/boolean_algrebra.json [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      },
+      {
+        "val": "Not",
+        "key": "¬"
+      }
+    ],
+	
+    "Basic Principles": [
+      {
+        "key": "X+0=X"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/boolean_algrebra.json]: I feel like we should space these out for legibility. We should put spaces around the `=` and `+` and other operators

**Pull request closed without merging**